recipes/astropy-base/meta.yaml

```diff
@@ -0,0 +1,83 @@
+{% set version = "6.1.3" %}
+
+package:
+  name: astropy-base
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/a/astropy/astropy-{{ version }}.tar.gz
```

Comment:
This should be another output on the astropy feedstock and not a separate recipe, it should also be a host and run dependency of astropy to avoid file clobbering

Reply:
I think multiple outputs may still be the best approach. You can have `skip: true` on some outputs for some combinations (eg python 3.13), so not all builds have to produce all outputs.

Reply:
My huge concern with that is migration, e.g. I don't expect it will receive automated PRs for e.g. py 3.13 migration until all the dependencies are resolved first. Surely a maintainer can go in and do it by hand by introducing a lot of conditionals and skips, but that kind of kills the purpose of automated migration. (and frankly, as one of the maintainers of the given feedstock, I don't see myself doing that; it will be a big burden and won't really serve the users either as suddenly astropy will become one of the packages to be migrated much later rather than being an early adopter (like it is with its official releases and CI setup).

Reply:
Well, the upstream feedstock has a built-in matrix happening, already, based on `os x arch x python x numpy`.

A way to avoid migration issues is to have all the upstream's `optional-dependencies` (which I _would_ encourage modeling the outputs after, rather than just kitchen-sinking) only be built on _exactly one_, known-good-and-fast build, e.g. `{linux-64,py310,numpy1.x}`

```yaml
# recipe/conda_build_config.yaml
astropy_outputs:
  - all    # [linux64 and py310]
  - base   # [not (linux64 and py310])]
```

then

```yaml
# recipe/meta.yaml
package:
  name: astropy-base

# ... pretty much everything that's already in there

outputs: 
  - name: astropy-base # this really is empty, and would be built for every permutation

{% if astropy_outputs = "all" %}
  - name: astropy
    build:
      noarch: generic
    # ... and then whatever else an _end user_ should expect, maybe `astropy-recommended`

  - name: astropy-ipython
    # ... and then whatever else it needs, including `pin_subpackage("astropy-base", max_pin="x.x.x")

  # ... all the other extras

  - name: astropy-all
  # ... depends on all the extras

{% endif %}
```

On that _one_ matrix intersection, it _will_ still have to solve all the output `run` + `test` environments during test, but not the `host`, and every _other_ build will be able to use  the packages generated during The One True Build, _provided_ all of _those_ (transient) `run` dependencies can be met in the user's environment.

Reply:
I think your analysis of automatic migrations is right, so if that's a big concern and prompt _automatic_ migrations are important and optional dependencies are likely to lag by an unacceptable margin, I do see the reasoning for a separate feedstock. I know I've made my share of manual migration PRs when I'm feeling impatient and I expect the migration will work anyway.

Looking at https://github.com/conda-forge/astropy-feedstock/pull/137, it does look like the new split `astropy` should become an empty metapackage with dependencies, which means it could be `noarch: generic` (i.e. it wouldn't need migration PRs at all). I'm not sure making this a noarch output in a single feedstock would affect the migration bots, but I doubt it and I'm not sure how to test.